### PR TITLE
Add new C++ mangle syntax.

### PR DESCRIPTION
**Pull request by look-at-me**
See https://github.com/dlang/dmd/pull/8667

### CODE REVIEW

**Comment by look-at-me** [comment]
Not sure if the CI builds the docs to look at? I couldn't build it on my system. The win32.mak just doesn't work and the posix.mak requires some specific folder layout from the looks of it. Was looking for a folder outside it's own `../dmd/*`.

File: spec/attribute.dd [hunk 1]
```diff
@@ -189,6 +195,40 @@ $(H3 C++ $(LNAME2 namespace, Namespaces))
 
         $(CPPCODE namespace N { namespace M { namespace R { void foo(); } } })
 
+        $(P The linkage form $(D extern $(LPAREN)C++, )$(I StringLiteralList)$(D $(RPAREN))
+        provides an alternative syntax which only modifies the mangle of the symbol.
+        No C++ namespace features are emulated allowing for the D module system to be used
+        unobstructed.
```
Comment (Geod24) [inline spec/attribute.dd]: Another advantage is that D keyword can be used

**Comment by look-at-me** [comment]
```
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on .generated/dlangspec.log.
posix.mak:544: recipe for target '/dev/shm/dtest/work/repo/dlang.org/web/dlangspec.pdf' failed
make: *** [/dev/shm/dtest/work/repo/dlang.org/web/dlangspec.pdf] Error 1
fatal: 'refs/ae-sys-d-cache/website-5648c9a45e689256e58981f3b450079e30eb50fd-f2baf3c634d68ccbd27bae18dea84bbd' - not a valid ref
```

The error messages are pretty horrible...

**Comment by jacob-carlborg** [comment]
Please squash commits when done

File: spec/attribute.dd [hunk 2]
```diff
@@ -189,6 +195,40 @@ $(H3 C++ $(LNAME2 namespace, Namespaces))
 
         $(CPPCODE namespace N { namespace M { namespace R { void foo(); } } })
 
+        $(P The linkage form $(D extern $(LPAREN)C++, )$(I StringLiteralList)$(D $(RPAREN))
+        provides an alternative syntax which only modifies the mangle of the symbol.
```
Comment (jacob-carlborg) [inline spec/attribute.dd]: Does it sound better with “the mangled name”?

**Comment by look-at-me** [comment]
Yah I don't know why this isn't building, the error message is useless and the CI takes forever to complete.

**Comment by thewilsonator** [comment]
@look-at-me that's due to a bad version of dub being used to build stuff by the CI, please rebase and it should go away.

**Comment by ZombineDev** [comment]
Rebased

**Comment by Geod24** [comment]
Could use a squash. Do you know why it fails ?

File: spec/attribute.dd [hunk 3]
```diff
@@ -77,6 +77,7 @@ $(GRAMMAR
 $(GNAME LinkageAttribute):
     $(D extern) $(D $(LPAREN)) $(GLINK LinkageType) $(D $(RPAREN))
     $(D extern) $(D $(LPAREN)) $(D C++), $(GLINK2 declaration, IdentifierList) $(D $(RPAREN))
+    $(D extern) $(D $(LPAREN)) $(D C++), $(GLINK StringLiteralList) $(D $(RPAREN))
```
Comment (Geod24) [inline spec/attribute.dd]: It also accepts an expression now, since @TurkeyMan 's PR was merged.

Comment (TurkeyMan) [reply spec/attribute.dd]: StringExp list...?

**Comment by Basile-z** [comment]
Close because obsolete. More or less the same has been added by another PR

File: spec/attribute.dd (hunk 3 above)
Comment (TurkeyMan) [reply spec/attribute.dd]: No?

**Pull request closed without merging**